src/core/src/bound_evaluate.cpp

```diff
@@ -247,7 +248,7 @@ bool ov::could_propagate(const Output<Node>& output, std::vector<Node*>& result)
             bool can_add = true;
             size_t arg_count = node->get_input_size();
 
-            if (arg_count == 0 && !is_type<op::v0::Constant>(node)) {
+            if ((arg_count == 0 && !is_type<op::v0::Constant>(node)) || is_decompression(node->shared_from_this())) {
```

Comment:
Convert operations with decompression attribute will be left untouched (won't be inserted in the result vector) which would result in absence of partial values in the output of Convert. This means that some operations will have insufficient shape propagation due to lack of constant data. Which also may result in non-optimal subgraphs due to transformations being disabled for unknown shapes.

Reply:
i see. i will try to check if the node is in shape subgraphs and if so do not disable evaluate for them

Reply:
Done. Now i do not skip evaluate and CF for shape path. Just to clarify only thing is that if there is e.g. the following graph
 ```mermaid
graph TD;
    ShapeOf-->subgraph_for_start;
    subgraph_for_start-->Range;
    stop-->Range;
    Range-->Convert_to_FP32;
    Convert_to_FP32-->MatMul;
    input_2-->MatMul;
```

All decompression converts inside `subgraph_for_start` will be evaluaated and folded. But evaluate for `Convert_to_FP32` will not be called, it will not be folded. This is not a bug but a feature: we don't want to convert such big size data tensors with slow ngraph evaluate, we want to do it on the plugin side with fast kernels.